modules: Add DevAlert (DFM) and move Tracerecorder into percepio module

aronlander-pe opened the pull request:
This commit adds the DFM (DevAlert target side code) module and moves the TraceRecorder module into the percepio module, which results in the TraceRecorder module definition being removed from the west manufest and module definition within zephyr.

aronlander-pe committed:
modules: Add DFM and move Tracerecorder into percepio module

This commit adds the DFM (DevAlert target side code) module and
moves the TraceRecorder module into the percepio module, which
results in the TraceRecorder module definition being removed
from the west manufest and module definition within zephyr.

Signed-off-by: Aron Lander <aron.lander@percepio.com>

eriktamlin reviewed (dismissed):
Swapped to new repo for TraceRecorder that also includes DevAlert (DFM)

aronlander-pe committed:
west.yml: Updated to use the latest version of the percepio module

The percepio module change removes unused kernel ports from the
zephyr version of the percepio module.

Signed-off-by: Aron Lander <aron.lander@percepio.com>
